### PR TITLE
Feature/8395 qrscanner/stick coordinators together

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Puts everything together to use the new universal QR code scanner in all places

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9428

### CODE REVIEW

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 1]
```diff
@@ -31,8 +31,10 @@ class ExposureSubmissionCoordinatorModel {
 
 	let exposureSubmissionService: ExposureSubmissionService
 	let coronaTestService: CoronaTestService
-
+	let eventProvider: EventProviding
+	
 	var coronaTestType: CoronaTestType?
+	var markAsUnseen: Bool = false
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift:37]: Hmm I don't unterstand what is marked as unseen here. 
Can you explain it to me or have a more self-explaining name?

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	case onBehalfFlow
+	case checkinTab
+	case certificateTab
+}
+
+class QRScannerCoordinator {
+	
+	// MARK: - Init
+	
+	init(
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift:19]: Hardcore DI 🤘

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift [hunk 3]
```diff
@@ -72,11 +78,10 @@ class QRScannerViewController: UIViewController {
 
 	private let focusView = QRScannerFocusView()
 	private let dismiss: () -> Void
-
+	private let flashButtonTag = 12
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:81]: Maybe nothing for now, but we have a Tag on a view to find it later on? 😳🙀

**Review by marcussc — APPROVED** [review]
Just some small questions/remarks otherwise LGTM

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]: Renamed it

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:81]: I'm just as surprised as you are 😆

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:81]: Already on it.

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:81]: I just was in a call with @naveeddotio  add asked him to change that.

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:81]: I just took the legacy wrote already there written by some legend :D

**Pull request merged**